### PR TITLE
Added new account service folder and new account login api

### PR DESCRIPTION
**Pull request by qinxutan**

### CODE REVIEW

File: account-service/backend/app.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from flask import Flask, request, jsonify, session
+from flask_cors import CORS
+from login import authenticate_user
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = '3f9a4c5b6d7e8f9a0b1c2d3e4f5g6h7i8j9k0l1m2n3o4p5q6r7s8t9u0v1w2x3y4z5a6b7c8d9e0f1g2h3i4j5k6l7m8n9o0p1q2r3s4t5u6v7w8x9y0z1a2b3c4d5e6f7g8h9i0j1k2l3m4n5o6p7q8r9s0t1u2v3w4x5y6z7a8b9c0d1e2f3g4h5i6j7k8l9m0n1o2p3q4r5s6t7u8v9w0x1y2z3a4b5c6d7e8f9g0h1i2j3k4l5m6n7o8p9q0r1s2t3u4v5w6x7y8z9'
```
Comment (Tan-Jia-Rong) [inline account-service/backend/app.py]: SECRET_KEY should be kept secure .-. 

Store it in a env variable and call it when needed

**Review by Tan-Jia-Rong — COMMENTED** [review]
whoa you is very efficient

Apart from that the login service should also gitignore the node modules perhaps use
`*/node_modules` and `node_modules` on the top level gitignore?

File: account-service/backend/app.py (hunk 1 above)
Comment (qinxutan) [reply account-service/backend/app.py]: OH OH OH thank u..

Comment (qinxutan) [reply account-service/backend/app.py]: Ok noted on the gitignore! will edit

File: account-service/backend/app.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from flask import Flask, request, jsonify, session
```
Comment (ekweirui) [inline account-service/backend/app.py]: I thought we were using node.js as the language and express as the framework for our backend.
Flask is a web framework similar to express, but it's for Python.

Perhaps we should convert these to js, unless we want to change our tech stack to python?
Sorry for only feedbacking this now after you have done all these up already 🥲

**Review by ekweirui — COMMENTED** [review]
Thanks for pushing your code for review so early! 
It's great to have a skeleton and some logic in place. 

However, one concern I have is the usage of python and flask in this review, most likely we will have to convert them to js and express.

File: account-service/backend/app.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from flask_cors import CORS
```
Comment (ekweirui) [inline account-service/backend/app.py]: I feel that moving the backend outside of the frontend would be better and provides for cleaner separation.

Using our tutorial skeleton as a context, the `demo-app` folder is our frontend and `demo-service` folder is our backend.

The rough skeleton for both the frontend and backend are already in their respectively folders, so we would just have to shift the code and logic from this review into the respectively folders/files.

For reference, from chatgpt, the rough directory structure of our repo should look like this:
image



And as for within the backend folder, the structure (including login logic from your review) should roughly look like this:
image



Please do bear in mind that their `frontend` folder = our `demo-app` folder, and their `backend` folder = our `demo-service` folder.

Sorry for asking you to perform a huge restructure.

File: account-service/backend/app.py (hunk 2 above)
Comment (qinxutan) [reply account-service/backend/app.py]: OH yep no worries!! i will convert them

File: account-service/backend/app.py (hunk 3 above)
Comment (qinxutan) [reply account-service/backend/app.py]: yep got it! will restrucutre accordingly

**Comment by qinxutan** [comment]
WIll create a new pr for this!

**Pull request closed without merging**